src/Aspire.Hosting/DistributedApplicationOptions.cs

```diff
@@ -101,7 +101,31 @@ internal string? ProjectName
     private string? ResolveProjectName()
     {
         var assemblyMetadata = Assembly?.GetCustomAttributes<AssemblyMetadataAttribute>();
-        return GetMetadataValue(assemblyMetadata, "AppHostProjectName");
+        var projectNameFromMetadata = GetMetadataValue(assemblyMetadata, "AppHostProjectName");
```

Comment:
@DamianEdwards how do we detect single file?

Reply:
Oh at runtime? Property in `AppContext`: `AppContext.GetData("EntryPointFilePath") as string`.

Example at https://github.com/DamianEdwards/runfile/blob/26436576c0fa0bb051c2a6b7f194f2b16c0346fc/flat/filepath.cs#L28